Feat/verifiable claims (#396)
okp4/docs: docs/academy/verifiable-claims.md

---
sidebar_class_name: hidden
---

# Verifiable Claims Principles

> <i>Reading time: {readingTime} min</i>

Author: It is a variable dynamically calculated.

In the [W3C standard](https://www.w3.org/TR/vc-data-model/), a credential is a set of one or more **claims** made by the same entity. Credentials include an identifier and metadata describing the credential's properties, such as the issuer, the expiry date and time, a representative image, a public key to use for verification purposes, the revocation mechanism, and so on. The issuer might sign the metadata. A verifiable credential is a set of tamper-evident claims and metadata that cryptographically prove who issued it.

<centre>
![verifiable-claims-1](/img/content/academy/verifiable-claims-1.webp)

> </centre>

Author: It doesn't work with Center instead of Centre.

> </centre>

Reviewer: @MdechampG Mixing HTML and Markdown syntax does not work. If you want to use the `<center/>` HTML tag, you must rewrite your code from Markdown to HTML. Below is a functional example extracted from the `technical-documentation` section (note that you can also manage the maximum width of the image if necessary):

```html
<center>
  <img src="/img/content/technical-documentation/governance-hierarchy-of-norms.webp" alt="Governance - Hierarchy of Norms" style={{ maxHeight: "340px" }}></img>
</center>
```

> </centre>

Reviewer: @coderabbitai, to make the centering function correctly, it is essential not only to change `<centre/>` to `<center/>` as you suggested but also to convert the content of the tag, which is in Markdown, into HTML using the `<img/>` tag.

> </centre>

Reviewer: @coderabbitai show me the fixed code. :relieved:

> </centre>

Reviewer: At first glance, it looks perfectly fine. Thanks. :+1:

A **verifiable presentation** is a tamper-evident presentation encoded in such a way that authorship of the data can be trusted after a process of cryptographic verification.

:::

### Example: Issuing a Digital Passport

A government authority issues a digital passport to a citizen.

<center>
  <img src="/img/content/academy/verifiable-claims-2.webp" alt="Verifiable claims passport example" style={{ maxHeight: "650px" }}></img>
</center>

Use when traveling :

- Airport Verification: When traveling, the passport holder presents his or her digital passport at the airport.
- Verifier (Customs): Customs officers use a system that verifies the passport's digital signature and metadata to confirm its authenticity.

The digital signature and metadata ensure enhanced security by preventing forgery and guaranteeing data integrity. Compliance with W3C standards facilitates fast, efficient data verification by verifiers.

## Verifiable claims in the OKP4 protocol

> The [OKP4 ontology](https://docs.okp4.network/academy/resources#ontology-how-its-represented-semantic-description) is used to describe the Dataverse. In the Dataverse, resources are identified by their DID. Resources are described by claims referenced in the ontology as Verifiable Credentials (VCs) whose subject is the resource's DID. In this way, each resource is described by a set of associated claims.

Reviewer: ```suggestion
The [OKP4 ontology](https://docs.okp4.network/academy/resources#ontology-how-its-represented-semantic-description) is used to describe the Dataverse. In the Dataverse, resources are identified by their DID. Resources are described by claims referenced in the ontology as Verifiable Presentations (VPs) whose subject is the resource's DID. In this way, each resource is described by a set of associated claims.
```

It's important to note that the existence of a resource is conditional on the presence of at least one relevant claim. In other words, although the universe of potential resources is infinite, a resource only becomes concrete and tangible when a corresponding claim is established.

:::note
This notion is analogous to that of crypto-currency wallets. Although all addresses theoretically exist, they only take on practical significance (such as indicating a balance) when participating in a transaction.
:::

Verifiable Credentials obey schemas that describe a particular aspect of a resource. Certain VCs apply depending on the nature of the resource. A VC describing an entity or organization does not apply to the same resource as a VC describing a dataset. This is why VCs are organized in schemas.

<center>
  <img src="/img/content/academy/verifiable-claims-3.webp" alt="Verifiable claims attached to a resource" style={{ maxHeight: "400px" }}></img>
</center>

### Resource materiality

> It has been established that the concrete existence of a resource depends on the presence of a claim concerning it. More specifically, **the Resource Governance VC ensures the existence of a resource in the protocol**: a resource or zone exists in the protocol only if governance is attached to it.

Reviewer: ```suggestion
It has been established that the concrete existence of a resource depends on the presence of a claim concerning it. More specifically, **the Resource Governance VC ensures the existence of a resource in the protocol**: a resource or zone exists in the protocol only if governance is attached to it.

```

These VCs play a crucial role in regulating and managing resources. They define the parameters for controlling, managing, and interacting with the resource and thus regulate its use, sharing, and modification in the dataverse. These VCs guarantee order and compliance with established resource management standards.

### Claim and revocation

#### Self-sovereign governance principles

> Resources are considered to be under "Self-Sovereign" management according to the following principles:

Reviewer: ```suggestion
Resources are considered to be under "Self-Sovereign" management according to the following two principles:

```

> Free Issuance of Claims: In this framework, any entity (individual, organization, service, etc.) can issue claims on the resource concerned, with the exception of governance claims, which are issued by the identities themselves.

Reviewer: ```suggestion
- *Free Issuance of Claims*: In this framework, any entity (individual, organization, service, etc.) can issue claims on the resource concerned, with the exception of governance claims, which are issued by the identities themselves.
```

Author: Changed thanks

> Free Issuance of Claims: In this framework, any entity (individual, organization, service, etc.) can issue claims on the resource concerned, with the exception of governance claims, which are issued by the identities themselves.

Reviewer: I think it's important to emphasise that this framework only applies to VCs that are not governance VCs. It's written, but not emphasised enough, imho.

Author: Indeed, I highlighted it.

> Unilateral revocation of claims: This principle stipulates that claims can only be revoked by the entity that issued them. This means that if Entity A issues a claim on the resource, only Entity A can revoke that claim. This principle is unalterable and remains independent of any intervention or modification by governance structures.

Reviewer: ```suggestion
- *Unilateral revocation of claims*: This principle stipulates that claims can only be revoked by the entity that issued them. This means that if Entity A issues a claim on the resource, only Entity A can revoke that claim. This principle is unalterable and remains independent of any intervention or modification by governance structures.

```

Self-sovereign governance is particularly well suited to environments where autonomy and decentralization are paramount, as is the case in the OKP4 dataverse. However, we must remain vigilant to the "gossip effect": given that any entity can issue credentials, there is a risk that unverified or inaccurate information will be disseminated. This phenomenon poses a significant challenge in balancing individual freedom and collective interest. Such dynamics need to be carefully considered when designing governance schemes in Zones.

## Registration of Verifiable Presentation in the OKP4 Protocol

Once the Issuer successfully issues the Verifiable Credentials (VCs) associated with a resource or a Zone to the Holder, the Registrant will record a Verifiable Presentation (VP) in the OKP4 blockchain. A VP consists of a set of VCs. The Registrant can be either the Holder or a third party.

Recording VPs in the blockchain, rather than individual VCs, ensures the confidentiality of specific properties by only revealing a restricted set of properties on the public chain.

<centre>
![verifiable-claims-4](/img/content/academy/verifiable-claims-4.webp)

> </centre>

Author: Done thank you
